app/src/main/java/com/duckduckgo/app/location/ui/SiteLocationPermissionDialog.kt

```diff
@@ -89,12 +108,18 @@ class SiteLocationPermissionDialog : DialogFragment() {
     private fun populateFavicon(imageView: ImageView) {
         arguments?.let { args ->
             val originUrl = args.getString(KEY_REQUEST_ORIGIN)
-            val faviconUrl = Uri.parse(originUrl).faviconLocation()
-
-            GlideApp.with(requireContext())
-                .load(faviconUrl)
-                .error(R.drawable.ic_globe_gray_16dp)
-                .into(imageView)
+            val tabId = args.getString(KEY_TAB_ID, "")
+
+            originUrl?.let { url ->
+                faviconJob?.cancel()
+                faviconJob = GlobalScope.launch {
```

Comment:
how about `viewLifecycleOwner.lifecycleScope.launch {  }` instead of `GlobalScope.launch{ }` ?

Reply:
Can't really use `viewLifecycleOwner` because at that point it crashes the app. I've changed it to `this.lifecycleScope`.